File: JavaAndScala/src/main/java/com/java/se/conclusion/design/factory/abstracts/WindowsFactory.java
package com.java.se.conclusion.design.factory.abstracts;

import com.java.se.conclusion.design.factory.abstracts.common.Border;
import com.java.se.conclusion.design.factory.abstracts.common.Button;
import com.java.se.conclusion.design.factory.abstracts.common.WindowsBorder;
import com.java.se.conclusion.design.factory.abstracts.common.WindowsButton;

/**
 * 	This is a class to implement abstract factory pattern
 *  -- This class must implement a parent interface
 *  
 * @author VinceYuan
 *
 */
public class WindowsFactory implements BrandFactory {

	@Override
	public Button getButton() {
		return new WindowsButton();
	}

	@Override
	public Border getBorder() {
		return new WindowsBorder();
	}
}

File: JavaAndScala/src/main/scala/org/lang/scala/conclusion/design/factory/abstracts/WindowsFactory.scala
package org.lang.scala.conclusion.design.factory.abstracts

import org.lang.scala.conclusion.design.factory.abstracts.common.WindowsBorder
import org.lang.scala.conclusion.design.factory.abstracts.common.WindowsButton

/**
 * 	This is a stand-alone class to implement abstract factory pattern
 *  -- This class must implement a parent trait
 *  
 * 	@author VinceYuan
 */
class WindowsFactory extends BrandFactory {
  	
	override def getButton() = new WindowsButton()
	override def getBorder() = new WindowsBorder()
}